state-reconstruct-fetcher/src/l1_fetcher.rs

```diff
@@ -274,11 +274,16 @@ impl L1Fetcher {
 
                     // Create a filter showing only `BlockCommit`s from the [`ZK_SYNC_ADDR`].
                     // TODO: Filter by executed blocks too.
+                    // Don't go beyond `end_block_number` - tip of the chain might still change.
+                    let mut last_l1_block_number = current_l1_block_number + BLOCK_STEP - 1;
+                    if last_l1_block_number > end_block_number {
+                        last_l1_block_number = end_block_number;
+                    }
```

Comment:
It's not really the `last_l1_block_number`; per iteration, we're incrementing by `BLOCK_STEP`. We could also replace the clamping logic with `std::cmp::min`, so how about:
```suggestion
                    let next_l1_block_number = cmp::min(
                        current_l1_block_number + BLOCK_STEP - 1,
                        end_block_number
                    );
```

Reply:
well, it's the last L1 block number in the filter; also, I'm not keen on `next_l1_block_number`, which is a name already used for something different (off by 1) below...

Reply:
OK